subsys/bluetooth/host_extensions/custom_ltk.c

```diff
@@ -0,0 +1,109 @@
+#include <bluetooth/nrf/host_extensions.h>
+#include <errno.h>
+#include <stdint.h>
+#include <string.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+
+#include "../subsys/bluetooth/host/keys.h"
+#include "../subsys/bluetooth/host/conn_internal.h"
+#include "zephyr/kernel.h"
+
+enum {
+	NRF_BT_KEYS_TRANSIENT = BIT(7),
+};
+
+int nrf_bt_conn_set_ltk(struct bt_conn *conn, const struct nrf_ltk *ltk, bool authenticated)
+{
+	struct bt_keys *keys;
+
+	CHECKIF(conn == NULL || ltk == NULL) {
+		return -EINVAL;
+	}
+
+	/* This hopefully a crude but decent prevention of any race
+	 * conditions with smp.c.
+	 *
+	 * It should prevent a race with smp_security_request(),
+	 * smp_security_request() and bt_smp_start_security(), which
+	 * will all allocate keys early and cause the following code
+	 * to find a key and return -EALREADY.
+	 */
+	k_sched_lock();
+
+	/* If there is a prior pairing for this address, we don't
+	 * allow setting the LTK manually. Please unpair first.
+	 *
+	 *
+	 */
+	keys = bt_keys_find(BT_KEYS_ALL, conn->id, &conn->le.dst);
+	if (keys) {
+		return -EALREADY;
```

Comment:
Maybe add a LOG_WRN here mentioning what is said in the comment above?
Something like "A prior pairing exists for %s (addr), please unpair first."